aesara/link/jax/jax_dispatch.py

```diff
@@ -184,9 +183,30 @@ def jax_func(*inputs):
     return jax_funcs
 
 
+@singledispatch
+def jax_typify(data, dtype):
+    """Convert instances of Aesara `Type`s to JAX types."""
+    if dtype is None:
+        return data
+    if dtype is not None:
+        return jnp.array(data, dtype=dtype)
+    raise NotImplementedError(f"No JAX conversion for data and dtype: {data}, {dtype}")
+
+
+@jax_typify.register(np.ndarray)
+def jax_typify_ndarray(data, dtype):
+    return jnp.array(data, dtype=dtype)
+
+
+@jax_typify.register(RandomState)
+def jax_typify_RandomState(state, dtype):
+    state = state.get_state()[1]
```

Comment:
We should use the newer NumPy API whenever possible (i.e. `state.get_state(legacy=False)["state"]`).

Reply:
Made the change in the latest commit.